rootconf/default/etc/init.d/discover.sh

```diff
@@ -68,6 +69,7 @@ do_start() {
             install_remain_sticky_arch || {
                 echo "Error: problems making install boot mode sticky" > /dev/console
             }
+            sync;sync
```

Comment:
Nice, I am glad this helps #250.

Instead of fixing it here, however, I would make this change in rootconf/x86_64/sysroot-lib-onie/boot-mode-arch.  This problem only effects x86, not powerpc or arm.